Receiver/test/test.cpp

```diff
@@ -0,0 +1,20 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+#include<iostream>
+#include <stdlib.h>
+#include <sstream>
+#include "../bms_receiver.h"
+#include "catch.hpp"
+
+TEST_CASE("data reception test") {
+  
+  bms_process_receiver testObj;
+  BMS_DATA* sampleData;
+  std::string teststring = "{'temperature': 15, 'soc': 59}";
+  REQUIRE(validate_data(teststring));
+  std::stringstream IPData;
+  IPData << "{'temperature': 15, 'soc': 59}";
```

Comment:
Nice test - it serves to document the input format too.
More tests would have documented your intent in a more complete way. Questions that a user of the receiver will have are: What happens if I add another sensor (charge cycles), which is reported before the temperature? What if I put it at the end? What if there is no space between the : and the number?
_It is not about handling all of these._ It is about defining a fixed behavior in unexpected situations. Often this makes the difference between a satisfied customer and an unhappy one.